### PR TITLE
Lab5 code updates

### PR DESCRIPTION
**Pull request by Carl-Hugo**
Here are the few updates that I made:

- Update `ImageAnalyzer` and `BlobStorageManager` to improve readability
- Update the way options are loaded to leverage the .net core 2.2 option
model. Move the options validation to their respective class using
`DataAnnotations`. Reorganize the Startup code to make it easier to follow.
- Add `deployment/Deploy.ps1` script to help other people get started quickly if needed
- Display the analyzed image at the bottom of the result page

Once the code is ok for you, I will look at updating the code snippets of the README file, so it reflect the changes.

### CODE REVIEW

File: Lab5/deployment/deploy.ps1 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[CmdletBinding()]
+param (
+    [Parameter(Mandatory = $true)][string]$webAppNameSuffix,
+    [string]$location = "canadacentral",
+    [string]$resourceGroupName = "GAB2019RG",
+    [string]$deploymentName = "Gab2019Deployment1",
+    [bool]$createResourceGroup = $false
```
Comment (DOMZE) [inline Lab5/deployment/deploy.ps1]: You could use a [switch] here instead...

File: Lab5/GABDemo/Services/ImageAnalyzer.cs [hunk 2]
```diff
@@ -20,18 +20,16 @@ public class ImageAnalyzer : IImageAnalyzer
...
         {
             if (!Uri.IsWellFormedUriString(imageUrl, UriKind.Absolute))
             {
-                throw new Exception("Invalid remoteImageUrl: {imageUrl}");
+                throw new Exception($"Invalid remoteImageUrl: {imageUrl}");
             }
-
-            ImageAnalysis analysis = await _computerVision.AnalyzeImageAsync(imageUrl, Features);
-            return analysis;
+            return _computerVision.AnalyzeImageAsync(imageUrl, Features);
```
Comment (DOMZE) [inline Lab5/GABDemo/Services/ImageAnalyzer.cs]: Personally, when I review, I prefer having the return statement split as it's easier for someone who wants to debug to put a breakpoint and see the properties of the variable. Refer to global comment for me feedback

**Comment by DOMZE** [comment]
Code is fine, there's many ways to go to Rome. Technically, we could argue all day about little nick nacks here and there. Everyone has their pet peeves. As a technical reviewer myself in my job, I tend to find the middle ground and not be a code zealot.

Merge if you believe it's good. The whole idea behind this is i want the people attending to learn and make it a smooth and easy experience

File: Lab5/GABDemo/Services/ImageAnalyzer.cs (hunk 2 above)
Comment (Carl-Hugo) [reply Lab5/GABDemo/Services/ImageAnalyzer.cs]: True enough (I think I said that to someone last week), I'll revert this to a "two-step return statement".

**Comment by Carl-Hugo** [comment]
> Code is fine, there's many ways to go to Rome. Technically, we could argue all day about little nick nacks here and there. Everyone has their pet peeves. As a technical reviewer myself in my job, I tend to find the middle ground and not be a code zealot.
> 
> Merge if you believe it's good. The whole idea behind this is I want the people attending to learn and make it a smooth and easy experience

For what its worth, my intention was not to be "a code zealot".
That said, the code is ok with me now and as you said, it is not a million dollars software 😉 . I will update the code samples from the README as soon as I can, and I will merge the PR.

**Comment by DOMZE** [comment]
Great amazing. Thanks for your contribution ;-)

**Pull request merged**